### PR TITLE
Run one executable for soca2cice (instead of two)

### LINKED ISSUE #1367: Consolidate soca -> cice increments into one run
Consolidate converting soca increments + backgrounds -> cice restarts into one task (both for arctic and antarctic).

### PR DESCRIPTION
**Pull request by shlyaeva**
# Description
Run a single executable to add soca increments to cice restart files, processing arctic and antarctic simultaneously to save on runtime and I/O.
Resolves https://github.com/NOAA-EMC/GDASApp/issues/1367

# Type of change
- [x] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? YES
  - [x] GDAS TBD

# How has this been tested?
- marine gdasapp tests on orion with intel

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [x] I have made corresponding changes to the system documentation if necessary

### CODE REVIEW

File: ush/python/pygfs/task/marine_analysis.py [hunk 1]
```diff
@@ -284,7 +284,7 @@ def _prep_checkpoint(self: Task) -> None:
 
         # render the SOCA to CICE YAML file for the Arctic and Antarctic
         logger.info("render the SOCA to CICE YAML file for the Arctic and Antarctic")
-        varchgyamls = ['soca_2cice_arctic.yaml', 'soca_2cice_antarctic.yaml']
+        varchgyamls = ['soca_2cice_global.yaml']
```
Comment (guillaumevernieres) [inline ush/python/pygfs/task/marine_analysis.py:287]: we could probably just get rid of the list and loop, unless we think we might be running more variable changes?

**Review by guillaumevernieres — DISMISSED** [review]
Thanks @shlyaeva !

**Comment by RussTreadon-NOAA** [comment]
@WalterKolczynski-NOAA , the GDASApp hash has been updated.  This PR is ready for CI testing and review.

**Comment by RussTreadon-NOAA** [comment]
GDASApp ctests, some of which include g-w jobs, [passed](https://github.com/NOAA-EMC/GDASApp/pull/1380#issuecomment-2492422059) on Hera.  See GDASApp PR [#1380](https://github.com/NOAA-EMC/GDASApp/pull/1380)

**Review by RussTreadon-NOAA — DISMISSED** [review]
Looks good to me.  Approve.

**Comment by RussTreadon-NOAA** [comment]
Since automated CI is not working for me, launch the following CI cases under my username on Hera
- _C96C48_hybatmDA_ as `prgsi_pr3118`
- _C96C48_ufs_hybatmDA_ as `prjedi_pr3118`
- _C96C48_hybatmaerosnowDA_ as `praero_pr3118`
- _C48mx500_3DVarAOWCDA_ as `prwcda_pr3118`
- _C96_atm3DVar_ as `pratm3dvar_pr3118`

**Review by WalterKolczynski-NOAA — DISMISSED** [review]

**Comment by RussTreadon-NOAA** [comment]
**Hera g-w CI results**
```
rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/prgsi_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202112201800        Done    Nov 22 2024 01:50:31    Nov 22 2024 02:15:22
202112210000        Done    Nov 22 2024 01:50:31    Nov 22 2024 05:25:15
202112210600        Done    Nov 22 2024 01:50:31    Nov 22 2024 04:50:15

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/prjedi_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202402231800        Done    Nov 22 2024 01:50:39    Nov 22 2024 02:25:23
202402240000        Done    Nov 22 2024 01:50:39    Nov 22 2024 06:05:22
202402240600        Done    Nov 22 2024 01:50:39    Nov 22 2024 06:10:20

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/praero_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202112201200      Active    Nov 22 2024 01:50:52             -
202112201800      Active    Nov 22 2024 01:50:52             -
202112210000      Active    Nov 22 2024 01:50:52             -

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/prwcda_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202103241200        Done    Nov 22 2024 01:50:54    Nov 22 2024 02:30:29
202103241800        Done    Nov 22 2024 01:50:54    Nov 22 2024 04:00:31

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/pratm3dvar_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202112201800        Done    Nov 22 2024 01:50:59    Nov 22 2024 02:30:30
202112210000        Done    Nov 22 2024 01:50:59    Nov 22 2024 05:45:26
202112210600        Done    Nov 22 2024 01:50:59    Nov 22 2024 04:55:20
```

All DA streams successfully completed except `praero_pr3118`.  

Job gdas_aeroanlgenb failed.  GDASApp issue [#1381](https://github.com/NOAA-EMC/GDASApp/issues/1381) has been opened to report this failure and document its resolution.

**Comment by RussTreadon-NOAA** [comment]
GDASApp PR [#1382](https://github.com/NOAA-EMC/GDASApp/pull/1382) commented out the section of GDASApp `parm/aero/berror/aero_diagb.yaml.j2` which caused the aeroanlgenb failure.  

@CoryMartin-NOAA commented 
> It will take a nontrivial amount of time to stage the rescaling file and make workflow changes such that it just made more sense to comment this out for now

Update `sorc/gdas.cd` in a working copy of `shlyaeva:feature/refactor_soca2cice` to bring in the updated GDASApp hash.  Now aeroanlgenb runs to completion.  Resume _C96C48_hybatmaerosnowDA_ running on Hera under my username.

**Review by AndrewEichmann-NOAA — APPROVED** [review]
Embrace The Simplifcation

**Review by RussTreadon-NOAA — APPROVED** [review]
C96C48_hybatmaerosnowDA works with the updated gdas.cd hash.

Approve.

**Review by guillaumevernieres — APPROVED** [review]
:+1:

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by RussTreadon-NOAA** [comment]
Confirm that `shlyaeva:feature/refactor_soca2cice` at bcc7ece1 can successfully run _C96C48_hybatmaerosnowDA_
```
rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/praero_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202112201200        Done    Nov 22 2024 01:50:52    Nov 22 2024 13:50:21
202112201800        Done    Nov 22 2024 01:50:52    Nov 22 2024 17:10:24
202112210000        Done    Nov 22 2024 01:50:52    Nov 22 2024 18:30:25
```

**Comment by WalterKolczynski-NOAA** [comment]
Ignore the failure, I'm fighting Jenkins this afternoon.

**Comment by RussTreadon-NOAA** [comment]
@WalterKolczynski-NOAA :  Where do we go from here?  

I repeated my build of this PR on Hera in `/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/pr3118`.  `build_all.log` successfully ran to completion
```
Hera(hfe11):/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/pr3118/sorc$ cat build_all.log
Running "module reset". Resetting modules to system default. The following $MODULEPATH directories have been removed: None
Building gsi_enkf, ufs, gfs_utils, gdas, ww3prepost, ufs_utils, gsi_utils, gsi_monitor, upp
Starting build_gsi_enkf.sh
Starting build_ufs.sh
Starting build_gfs_utils.sh
Starting build_gdas.sh
Starting build_ww3prepost.sh
Starting build_ufs_utils.sh
Starting build_gsi_utils.sh
Starting build_gsi_monitor.sh
Starting build_upp.sh
build_gfs_utils.sh completed successfully!
build_ufs_utils.sh completed successfully!
build_gsi_utils.sh completed successfully!
build_gsi_monitor.sh completed successfully!
build_ww3prepost.sh completed successfully!
build_upp.sh completed successfully!
build_gsi_enkf.sh completed successfully!
build_ufs.sh completed successfully!
build_gdas.sh completed successfully!

 .... Build system finished ....
```

I do not know the intricacies of Jenkins but it seems to have frequent hiccups when running g-w CI.  What can we do to make Jenkins based CI more robust?

I have greater success manually running `./workflow/create_experiment.py` to set up CI streams and then enabling a cron to rocotorun each stream to completion.  The following g-w CI configurations

- _C96C48_hybatmDA_ as `prgsi_pr3118`
- _C96C48_ufs_hybatmDA_ as `prjedi_pr3118`
- _C96C48_hybatmaerosnowDA_ as `praero_pr3118`
- _C48mx500_3DVarAOWCDA_ as `prwcda_pr3118`
- _C96_atm3DVar_ as `pratm3dvar_pr3118`

successfully ran for this PR via this approach.

The following CI cases remain to be run on Hera
- _C48_ATM_
- _C48_S2SW_

I assume we do not need to run the GEFS CI cases for this PR
- _C48_S2SWA_gefs_
- _C96_S2SWA_gefs_replay_ics_

I went ahead and manually set up _C48_ATM_ and _C48_S2SW_ for this PR.  Both streams are now running on Hera via cron.  Assuming these two streams pass, is their success along with success of the other five streams sufficient to allow this PR to be merged into `develop`?

Does g-w CI need to be run on other platforms?

**Comment by RussTreadon-NOAA** [comment]
_C48_ATM_ and _C48_S2SW_ are complete.
```
rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/pratm_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202103231200        Done    Nov 23 2024 12:29:20    Nov 23 2024 13:50:16
202103231800        Done    Nov 23 2024 12:29:20    Nov 23 2024 14:00:20

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/prs2sw_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202103231200        Done    Nov 23 2024 12:29:18    Nov 23 2024 15:30:27
202103231800        Done    Nov 23 2024 12:29:18    Nov 23 2024 15:36:20
```
All jobs in both streams successfully ran to completion

**Comment by RussTreadon-NOAA** [comment]
**Here g-w CI**
The following g-w CI has been run and passed on Hera
```
rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/prgsi_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202112201800        Done    Nov 22 2024 01:50:31    Nov 22 2024 02:15:22
202112210000        Done    Nov 22 2024 01:50:31    Nov 22 2024 05:25:15
202112210600        Done    Nov 22 2024 01:50:31    Nov 22 2024 04:50:15

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/prjedi_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202402231800        Done    Nov 22 2024 01:50:39    Nov 22 2024 02:25:23
202402240000        Done    Nov 22 2024 01:50:39    Nov 22 2024 06:05:22
202402240600        Done    Nov 22 2024 01:50:39    Nov 22 2024 06:10:20

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/praero_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202112201200        Done    Nov 22 2024 01:50:52    Nov 22 2024 13:50:21
202112201800        Done    Nov 22 2024 01:50:52    Nov 22 2024 17:10:24
202112210000        Done    Nov 22 2024 01:50:52    Nov 22 2024 18:30:25

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/prwcda_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202103241200        Done    Nov 22 2024 01:50:54    Nov 22 2024 02:30:29
202103241800        Done    Nov 22 2024 01:50:54    Nov 22 2024 04:00:31

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/pratm3dvar_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202112201800        Done    Nov 22 2024 01:50:59    Nov 22 2024 02:30:30
202112210000        Done    Nov 22 2024 01:50:59    Nov 22 2024 05:45:26
202112210600        Done    Nov 22 2024 01:50:59    Nov 22 2024 04:55:20

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/prs2sw_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202103231200        Done    Nov 23 2024 12:29:18    Nov 23 2024 15:30:27
202103231800        Done    Nov 23 2024 12:29:18    Nov 23 2024 15:36:20

rocotostat /scratch1/NCEPDEV/stmp2/Russ.Treadon/EXPDIR/pratm_pr3118
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202103231200        Done    Nov 23 2024 12:29:20    Nov 23 2024 13:50:16
202103231800        Done    Nov 23 2024 12:29:20    Nov 23 2024 14:00:20
```

**Comment by RussTreadon-NOAA** [comment]
Automated CI passed on Hera.  Yeah!

**Comment by RussTreadon-NOAA** [comment]
@WalterKolczynski-NOAA :  Is the Hera success sufficient?  If not, on what other machines should I run?

**Comment by RussTreadon-NOAA** [comment]
Thank you @WalterKolczynski-NOAA !

**Comment by shlyaeva** [comment]
Thank you all so much for very quick reviews, testing and turnaround!

**Pull request merged**